src/condor_includes/condor_crypt.h

```diff
@@ -26,6 +26,27 @@
 
 #include "CryptKey.h"
 
+struct StreamCryptoState {
+    // The IV is split in two: a 32-bit counter and a
+    // 64-bit random number for a total of 12 bytes.
+    // The ctr is added to the last 4 bytes of the IV.
+    union Packed_IV {
+        unsigned char iv[16]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
+        struct {
+            uint32_t pkt;
+            uint32_t conn;
+        } ctr;
+    };
+
+    uint32_t m_ctr_enc{0}; // Number of outgoing (encrypted) packets
+    uint32_t m_ctr_dec{0}; // Number of incoming (decrypted) packets
+    uint32_t m_ctr_conn{0}; // Number of times this session has been used by a connection
+    union Packed_IV m_iv_enc; // IV for outgoing data
+    union Packed_IV m_iv_dec; // IV for incoming data.
+    unsigned char m_prev_mac_enc[16]{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```

Comment:
Add comment for what these MACs are for.

Reply:
We don't actually need these MACs anymore if we're not going to add them to the AAD, right? (I removed that code because we have the IV counter, so we are still guarded against knockout attacks.)   Maybe they are useful for debugging, if you are trying to chain one message to the next, but you can do that with the last 8 bytes of the IV, so I'm tempted to remove.

Reply:
Correct - these are currently write-only structures.  Let's nuke 'em.

Reply:
Already gone...